src/firebolt/common/base_cursor.py

```diff
@@ -196,13 +198,16 @@ class BaseCursor:
 
     default_arraysize = 1
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
+    def __init__(
+        self, *args: Any, escape_chars: Dict[str, str] = {}, **kwargs: Any
+    ) -> None:
         self._arraysize = self.default_arraysize
         # These fields initialized here for type annotations purpose
         self._rows: Optional[List[List[RawColType]]] = None
         self._descriptions: Optional[List[Column]] = None
         self._statistics: Optional[Statistics] = None
         self._row_sets: List[RowSet] = []
+        self._formatter = StatementFormatter(escape_chars)
```

Comment:
Let's pass formatter directly to constructor instead of `escape_chars`. escape_chars is a part of internal Formatter behavior, and should not be exposed.
Makes sense to even incapsulate contruction of formatter_v1 and formatter_v2 into functions somewhere near the Formatter class definition
Though it's just a nit